lib/src/wallet.rs

```diff
@@ -1472,7 +1474,8 @@ impl LightWallet {
                     .iter_mut()
                     .find(|u| utxo.txid == u.txid && utxo.output_index == u.output_index)
                     .unwrap();
-                spent_utxo.unconfirmed_spent = Some((transaction.txid(), u32::from(last_height)));
+                spent_utxo.unconfirmed_spent =
+                    Some((transaction.txid(), u32::from(submission_height)));
```

Comment:
Are we correct in cases where the submitted note is pushed out of the mempool without making it into a block?